pwa/components/repairers/registration/RegistrationTunnelWorkshop.tsx

```diff
@@ -0,0 +1,466 @@
+  };
+
+  useEffect(() => {
+    if (bikeTypes.length === 0) {
+      fetchBikeTypes();
+    }
+  }, [bikeTypes]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const handleChangeName = (event: ChangeEvent<HTMLInputElement>): void => {
+    setName(event.target.value);
+  };
+
+  const fetchCitiesResult = useCallback(
+    async (cityStr: string) => {
+      const citiesResponse = await searchCity(cityStr, useNominatim);
+      const cities: City[] = useNominatim
+        ? createCitiesWithNominatimAPI(citiesResponse as NominatimCity[])
+        : createCitiesWithGouvAPI(citiesResponse as GouvCity[]);
+      setCitiesList(cities);
+    },
+    [setCitiesList]
+  );
+
+  useEffect(() => {
+    if (cityInput === '' || cityInput.length < 3) {
+      setCitiesList([]);
+    } else fetchCitiesResult(cityInput);
+  }, [setCitiesList, fetchCitiesResult, cityInput]);
+
+  const handleCityChange = async (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): Promise<void> => {
+    setCityInput(event.target.value);
+  };
+
+  const handleItinerantCityChange = (value: string): void => {
+    if (value === '' || value.length < 3) {
+      setCitiesList([]);
+    } else fetchCitiesResult(value);
+  };
+
+  const handleCitySelect = (
+    event: SyntheticEvent<Element, Event>,
+    cities: any
+  ) => {
+    let newRepairerCities: RepairerCity[] = [];
+
+    cities.map((city: RepairerCity | City) => {
+      if ('lat' in city && 'lon' in city) {
+        newRepairerCities.push({
+          formatted_name: `${city.name} (${city.postcode})`,
+          latitude: city.lat.toString(),
+          longitude: city.lon.toString(),
+          name: city.name,
+          postcode: city.postcode,
+        } as RepairerCity);
+      } else {
+        newRepairerCities.push(city as RepairerCity);
+      }
+    });
+
+    setRepairerCities(newRepairerCities);
+  };
+
+  const handleChangeStreetNumber = (
+    event: ChangeEvent<HTMLInputElement>
+  ): void => {
+    setStreetNumber(event.target.value);
+  };
+
+  const handleChangeStreet = async (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): Promise<void> => {
+    const adresseSearch = event.target.value;
+    if (adresseSearch.length >= 3) {
+      const streetApiResponse = await searchStreet(adresseSearch, city);
+      setStreetList(streetApiResponse);
+    }
+  };
+  const handleChangeComments = (event: ChangeEvent<HTMLInputElement>): void => {
+    setComment(event.target.value);
+  };
+
+  const handleChangeRepairerType = (
+    event: SelectChangeEvent<typeof repairerTypeSelected>
+  ): void => {
+    const {
+      target: {value},
+    } = event;
+    setRepairerTypeSelected(
+      typeof value === 'string' ? value.split(',') : value
+    );
+  };
+
+  useEffect(() => {
+    if (repairerTypeSelected.length > 0) {
+      setIsRoving(
+        repairerTypeSelected.some(
+          (element) => element === 'Réparateur itinérant'
+        )
+      );
+    }
```

Comment:
Why use `useEffect` here? I wonder if this code can be put in `handleChangeRepairerType` function?

Reply:
I putted it in a useEffect() because there is a delay when I do a setRepairerTypeSelected and the values can be wrong